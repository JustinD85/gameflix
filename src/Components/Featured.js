import React, { Component } from 'react';

export default class Featured extends Component {
  render() {

    return (
      <nav className="Featured">
        <iframe
          width="100%"
          height="100%"
          src={`https://www.youtube.com/embed/${this.props.data[1].youtube}?autoplay=0`}
          frameBorder="0"
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen
          title="featured-video">
        </iframe>
      </nav>
    );
  }
}