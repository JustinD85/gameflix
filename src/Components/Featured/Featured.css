.Featured {
  height: 33.33%;
  width: 100%;
  background-color: yellow;
}