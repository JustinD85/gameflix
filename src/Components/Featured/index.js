import React, { Component } from 'react';
import './Featured.css';

const Featured = (props) => {
  return (
    <nav className="Featured">
      <iframe width="100%" height="100%" src={"https://www.youtube.com/embed/" + props.data[1].youtube} frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
    </nav>
  )
}

export default Featured;