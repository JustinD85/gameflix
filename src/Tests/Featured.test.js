import React from 'react';
import { shallow } from 'enzyme';
import Featured from '../Components/Featured';

describe('Featured', () => {
  let wrapper;

  const games = [{}, {
    "youtube": "ZYvvhOzYx_4"
  }];

  beforeEach(() => {
    wrapper = shallow(
      <Featured data={games} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }) 

})