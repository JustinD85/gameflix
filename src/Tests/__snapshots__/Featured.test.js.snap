// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Featured
    data={
      Array [
        Object {},
        Object {
          "youtube": "ZYvvhOzYx_4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <iframe
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen={true}
        frameBorder="0"
        height="100%"
        src="https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0"
        title="featured-video"
        width="100%"
      />,
      "className": "Featured",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "allow": "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",
        "allowFullScreen": true,
        "frameBorder": "0",
        "height": "100%",
        "src": "https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0",
        "title": "featured-video",
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": "iframe",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          height="100%"
          src="https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0"
          title="featured-video"
          width="100%"
        />,
        "className": "Featured",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "allow": "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",
          "allowFullScreen": true,
          "frameBorder": "0",
          "height": "100%",
          "src": "https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0",
          "title": "featured-video",
          "width": "100%",
        },
        "ref": null,
        "rendered": null,
        "type": "iframe",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
